### ISSUE TITLE
Source Detection is restructured

### ISSUE THREAD

**Issue by kenfus** [i4Ds/Karabo-Pipeline#528]
Currently, Source Detection is quite wierd:

- you call `detected sources in image` on a class called `SourceDetectionResult`, which returns itself, but this time with the detected sources as an attribute. 
- Usage of `cls` makes maybe the code more clean but overall less readable.

Thing should be split:
- One class called `SourceDetection`, which results `SourceDetectionResult` with the results.

**Comment by Lukas113** [comment]
I don't agree with the first part of your complaints. using cls is common and better practice than using the class itself to allow inheritance.

However, I agree to split the classes in `SourceDetection` and `SourceDetectionResult`.

If cleanup is still an issue, should also be handled properly.
